include/openssl/ssl.h

```diff
@@ -1558,6 +1558,27 @@ __owur char *SSL_get_srp_username(SSL *s);
 __owur char *SSL_get_srp_userinfo(SSL *s);
 # endif
 
+/*
+ * Early callback and helpers.
+ */
+typedef struct clienthello_msg_st SSL_EARLY;
+typedef int (*ssl_early_cb_fn) (SSL *s, SSL_EARLY *ectx,
+                                const unsigned char *clienthello, size_t ch_len,
+                                int *al, void *arg);
+void SSL_CTX_set_early_cb(SSL_CTX *c, ssl_early_cb_fn cb, void *arg);
```

Comment:
Changed to using the majuscule in the typedef name.
There ... probably is not a need for a SSL_set_early_cb(), though I'm happy to put one in if you want.  Basically, the early callback itself is the first time you will get any data that could possibly make something unique per SSL derived from the same parent SSL_CTX, AIUI.  So, there isn't a "legitimate" need for it, just possibly one of convenience in some application coding styles.